oaknational/Oak-Web-Application
# feat: firebase auth

## Description

- adds firebase auth for authentication+session management
- enables email + magic link authentication
- adds pages:
  - `/sign-in`
  - `/sign-in/callback`
  - `/sign-in/error`
  - `/sign-in/success`
- adds `/firebase/functions` to add custom claims to jwt (for hasura)
  

## Issue(s)

Fixes #37

## How to test

1. Go to {cloud link}
2. Enter your email and click "Sign up/sign in"
3. Go to your email and click the link
4. You should see a success message
5. Navigate to the original tab
6. You should see a message saying "You are signed in as {email}"
7. Click "Sign out"
8. It should sign you out

## Screenshots

How it should now look:
![image](https://user-images.githubusercontent.com/12934669/159738187-223ea42d-2504-4d11-a9a3-e0188dcb0453.png)
![image](https://user-images.githubusercontent.com/12934669/159738243-7ccdc5e4-b55b-4283-bf55-4f3cfbb980ee.png)
![image](https://user-images.githubusercontent.com/12934669/159738378-41575db7-5626-437a-a019-d5354339b314.png)
![image](https://user-images.githubusercontent.com/12934669/159738423-ba8261a6-1538-4769-9465-ead8f7538a97.png)
